hazelcast/src/main/java/com/hazelcast/query/Predicates.java

```diff
@@ -38,68 +38,215 @@
     private Predicates() {
     }
 
+    /**
+     * Creates a new instance of <b>instance of</b> predicate.
```

Comment:
I would use `Creates a new instance of {@link InstanceOfPredicate}.` as the first line, same for all the other methods. Also use `{@link ...}` for the `@return`. I think it's always nice to have a clickable link in the JavaDoc.

It will also help to reduce the `instance of <b>instance of</b>` duplication in this case.

Reply:
It's private class. Also, Predicates are immutable I guess, so the "new instance" isn't important. Maybe:
> Creates a predicate that will pass items which are `instanceof` the given `klass`.
Btw, why there's no `attribute` parameter here?

Reply:
@Donnerbart providing links to the implementations was my first attempt on the issue, I agree with @viliam-durina, it's not a good idea to provide links to the internal classes.

@viliam-durina `InstanceOfPredicate` doesn't support `Extractable`, there is no point in having `attribute` parameter here.